assets/js/components/surveys/CurrentSurvey/utils.test.js

```diff
@@ -0,0 +1,219 @@
+				{
+					question_ordinal: 1,
+					answer: {
+						answer: [
+							{ answer_ordinal: 1 },
+							{ answer_ordinal: 2 },
+						],
+					},
+				},
+			];
+
+			const { currentQuestion, currentQuestionOrdinal } =
+				getCurrentQuestionAndOrdinal( questions, answers );
+
+			// Verify that the second question is returned since the answers include 1 and 2.
+			expect( currentQuestion ).toBe( questions[ 1 ] );
+			expect( currentQuestionOrdinal ).toBe( 2 );
+		} );
+	} );
+
+	describe( 'getTriggeredCompletion', () => {
+		it( 'should return undefined when the survey is incomplete', () => {
+			const { question: questions, completion: completions } =
+				singleQuestionSurvey.survey_payload;
+			const answers = [];
+			const currentQuestionOrdinal = 1;
+
+			const triggeredCompletion = getTriggeredCompletion(
+				questions,
+				answers,
+				completions,
+				currentQuestionOrdinal
+			);
+
+			expect( triggeredCompletion ).toBeUndefined();
+		} );
+
+		it( 'should return the matching completion when trigger conditions are met', () => {
+			const { question: questions, completion: completions } =
+				singleQuestionSurvey.survey_payload;
+			const answers = [
+				{
+					question_ordinal: 1,
+					answer: { answer: { answer_ordinal: 5 } },
+				},
+			];
+			const currentQuestionOrdinal = 2; // Past the last question.
+
+			const triggeredCompletion = getTriggeredCompletion(
+				questions,
+				answers,
+				completions,
+				currentQuestionOrdinal
+			);
+
+			expect( triggeredCompletion ).toBe( completions[ 0 ] );
+		} );
+
+		it( 'should return the first completion when no trigger conditions are met', () => {
+			const { question: questions, completion: completions } =
+				singleQuestionSurvey.survey_payload;
+			const answers = [
+				{
+					question_ordinal: 1,
+					answer: { answer: { answer_ordinal: 6 } },
+				},
+			];
+			const currentQuestionOrdinal = 2; // Past the last question.
+
+			const triggeredCompletion = getTriggeredCompletion(
+				questions,
+				answers,
+				completions,
+				currentQuestionOrdinal
+			);
+
+			expect( triggeredCompletion ).toBe( completions[ 0 ] );
+		} );
+	} );
+
+	describe( 'isLastQuestion', () => {
+		it( 'should return true for a single question survey', () => {
+			const questions = singleQuestionSurvey.survey_payload.question;
+			const currentQuestion = questions[ 0 ];
+
+			expect( isLastQuestion( questions, currentQuestion ) ).toBe( true );
+		} );
+
+		it( 'should return false for questions referenced by trigger conditions', () => {
+			const questions =
+				multiQuestionConditionalSurvey.survey_payload.question;
+			const firstQuestion = questions[ 0 ];
+
+			expect( isLastQuestion( questions, firstQuestion ) ).toBe( false );
+		} );
+
+		it( 'should return true for questions not referenced by trigger conditions', () => {
+			const questions =
+				multiQuestionConditionalSurvey.survey_payload.question;
+			const lastQuestion = questions[ questions.length - 1 ];
```

Comment:
A bit more readable?
```suggestion
			const lastQuestion = questions.at( -1 );
```

Reply:
Hmm, `at()` doesn't seem to be available:

![image](https://github.com/user-attachments/assets/0d51a967-eb2e-4086-8302-1f33fe298ba0)